### ISSUE TITLE
Token check is using parseFloat

### ISSUE THREAD

**Issue by apellizzn** [yeojz/otplib#96]
***Actual behavior***
If i call the function `verify` with a valid token it returns `true`

```
const token = "1234"
authenticator.verify({ token, secret }))
```
If i call the function `verify` with a valid token followed by any letter it returns `true`
```
const token = "1234abc"
authenticator.verify({ token, secret }))
```
If i call the function `verify` with a any letter followed by a valid token it returns `false`
```
const token = "abc1234"
authenticator.verify({ token, secret }))
```

**Expected behavior**
The expected behavior is to return `true` only in the first case

**Additional context**
Digging into the lib code i have found that the token comparison between the one provided by the user and the one generated by the lib is made using the following
```
function isSameToken(token1, token2) {
  return parseFloat(token1) === parseFloat(token2);
}
```
Applying `parseFloat` to `1234abc` will return `1234`, applying it to `acd1234` will rerturn `NaN` i think that in both cases `1234abc` and `acd1234` the function should return `false` since the  provided token is actually not the same as the one generated by the lib

**Comment by yeojz** [comment]
Hi @apellizzn 

For tokens, traditionally it should be integers only.
I think it will be better if it returns false the moment a token is `NaN` instead.
What do you think?

Good catch nonetheless!

**Comment by apellizzn** [comment]
Hi @yeojz i think that here the code should be just a mere string comparison, there are a lot of edge cases to test like what happens if i pass an octal `0o123.toString() === "83".toString()` :D
so probably i would implement it like this
```
function isSameToken(token1, token2) {
  return token1.toString() === token2.toString();
}
```

**Comment by yeojz** [comment]
fix released in v10.0.1

**Comment by yeojz** [comment]
reopening issue. release pipeline error-ed out.
